src/experimental/riff-raff-yaml-file/index.test.ts

```diff
@@ -540,4 +540,145 @@ describe("The RiffRaffYamlFileExperimental class", () => {
+          userData: {
+            distributable: {
+              fileName: `my-api.deb`,
+              executionStatement: `dpkg -i /my-api/my-api.deb`,
+            },
+          },
+          certificateProps: {
+            domainName: "api.devx.gutools.co.uk",
+          },
+          monitoringConfiguration: { noMonitoring: true },
+          scaling: {
+            minimumInstances: 1,
+          },
+          imageRecipe: "arm64-bionic-java11-deploy-infrastructure",
+        });
+
+        new GuNodeApp(this, {
+          app: "my-data-collector",
+          instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.MICRO),
+          access: { scope: AccessScope.PUBLIC },
+          userData: {
+            distributable: {
+              fileName: `my-data-collector.deb`,
+              executionStatement: `dpkg -i /my-data-collector/my-data-collector.deb`,
+            },
+          },
+          certificateProps: {
+            domainName: "data-collector.devx.gutools.co.uk",
+          },
+          monitoringConfiguration: { noMonitoring: true },
+          scaling: {
+            minimumInstances: 1,
+          },
+          imageRecipe: "arm64-bionic-node18-deploy-infrastructure",
+        });
+      }
+    }
+
+    new MyApplicationStack(app, "test-stack", { stack: "test", stage: "CODE", env: { region: "eu-west-1" } });
+
+    const actual = new RiffRaffYamlFileExperimental(app).toYAML();
+
+    expect(actual).toMatchInlineSnapshot(`
+      "allowedStages:
+        - CODE
+      deployments:
+        asg-upload-eu-west-1-test-my-api:
+          type: autoscaling
+          actions:
+            - uploadArtifacts
+          regions:
+            - eu-west-1
+          stacks:
+            - test
+          app: my-api
+          parameters:
+            bucketSsmLookup: true
+            prefixApp: true
+          contentDirectory: my-api
+        asg-upload-eu-west-1-test-my-data-collector:
+          type: autoscaling
+          actions:
+            - uploadArtifacts
+          regions:
+            - eu-west-1
+          stacks:
+            - test
+          app: my-data-collector
+          parameters:
+            bucketSsmLookup: true
+            prefixApp: true
+          contentDirectory: my-data-collector
+        cfn-eu-west-1-test-my-application-stack:
+          type: cloud-formation
+          regions:
+            - eu-west-1
+          stacks:
+            - test
+          app: my-application-stack
+          contentDirectory: /tmp/cdk.out
+          parameters:
+            templateStagePaths:
+              CODE: test-stack.template.json
+            amiParametersToTags:
+              AMIMyapi:
+                BuiltBy: amigo
+                Recipe: arm64-bionic-java11-deploy-infrastructure
+                AmigoStage: PROD
+              AMIMydatacollector:
+                BuiltBy: amigo
+                Recipe: arm64-bionic-node18-deploy-infrastructure
+                AmigoStage: PROD
```

Comment:
This is the fixed behaviour. Previously this would read:

```yaml
parameters:
  templateStagePaths:
    CODE: test-stack.template.json
  amiParameter: AMIMydatacollector
  amiTags:
    BuiltBy: amigo
    Recipe: arm64-bionic-node18-deploy-infrastructure
    AmigoStage: PROD
```

Information for `AMIMyapi` would simply be lost as it's last write wins. Use of `amiParametersToTags` allows us to describe multiple `amiParameter`s.